File: Examples/test-suite/javascript/preproc_defined_runme.js
var preproc_defined = require("preproc_defined");

if (preproc_defined.call_checking() != 1) {
    throw new Error;
}

d = new preproc_defined.Defined();
d.defined = 10;

preproc_defined.thing(10);
preproc_defined.stuff(10);
preproc_defined.bumpf(10);

if (preproc_defined.a != 2) {
    throw new Error;
}

if (preproc_defined.b != 42) {
    throw new Error;
}

if (preproc_defined.z != 8) {
    throw new Error;
}

File: Examples/test-suite/lua/preproc_defined_runme.lua
local v=require("preproc_defined")

assert(v.call_checking() == 1)
local d = v.Defined()
d.defined = 10

v.thing(10)
v.stuff(10)
v.bumpf(10)
assert(v.a == 2)
assert(v.b == 42)
assert(v.z == 8)
